src/QtLocationPlugin/EsriMapProvider.cpp

```diff
@@ -1,61 +1,51 @@
+/****************************************************************************
+ *
+ *   (c) 2009-2019 QGROUNDCONTROL PROJECT <http://www.qgroundcontrol.org>
+ *
+ * QGroundControl is licensed according to the terms in the file
+ * COPYING.md in the root of the source code directory.
+ *
+ ****************************************************************************/
+
 #include "EsriMapProvider.h"
 #include "QGCApplication.h"
-#include "QGCMapEngine.h"
 #include "SettingsManager.h"
 
-QNetworkRequest
-EsriMapProvider::getTileURL(int x, int y, int zoom,
-                            QNetworkAccessManager* networkManager) {
+EsriMapProvider::EsriMapProvider(const quint32 averageSize, const QGeoMapType::MapStyle mapType, QObject *parent)
+    : MapProvider(QString(), QString(), averageSize, mapType, parent) {}
+
+QNetworkRequest EsriMapProvider::getTileURL(const int x, const int y, const int zoom, QNetworkAccessManager* networkManager) {
     //-- Build URL
     QNetworkRequest request;
-    QString         url = _getURL(x, y, zoom, networkManager);
+    const QString url = _getURL(x, y, zoom, networkManager);
     if (url.isEmpty()) {
         return request;
     }
     request.setUrl(QUrl(url));
-    request.setRawHeader("Accept", "*/*");
-    QByteArray token = qgcApp()
-                           ->toolbox()
-                           ->settingsManager()
-                           ->appSettings()
-                           ->esriToken()
-                           ->rawValue()
-                           .toString()
-                           .toLatin1();
-    request.setRawHeader("User-Agent",
-                         QByteArrayLiteral("Qt Location based application"));
-    request.setRawHeader("User-Token", token);
+    request.setRawHeader(QByteArrayLiteral("Accept"), QByteArrayLiteral("*/*"));
+    const QByteArray token = qgcApp()->toolbox()->settingsManager()->appSettings()->esriToken()->rawValue().toString().toLatin1();
+    request.setRawHeader(QByteArrayLiteral("User-Agent"), QByteArrayLiteral("Qt Location based application"));
+    request.setRawHeader(QByteArrayLiteral("User-Token"), token);
     return request;
 }
 
-QString
-EsriWorldStreetMapProvider::_getURL(int x, int y, int zoom,
-                                    QNetworkAccessManager* networkManager) {
-    Q_UNUSED(networkManager);
-    return QString("http://services.arcgisonline.com/ArcGIS/rest/services/"
-                   "World_Street_Map/MapServer/tile/%1/%2/%3")
-        .arg(zoom)
-        .arg(y)
-        .arg(x);
+static const QString WorldStreetMapUrl = QStringLiteral("http://services.arcgisonline.com/ArcGIS/rest/services/World_Street_Map/MapServer/tile/%1/%2/%3");
```

Comment:
Why are you moving all URLs out of class scope ?

Reply:
Just to avoid allocating 2 objects at every request instead of just one (because arg always returns a *copy* of the initial string). You can as well add them in the class as static members (in the header).